### PR TITLE
Feature: Collaborators and assignees

### PR DESCRIPTION
**Pull request by FranzPoize**

### CODE REVIEW

File: resolvers/rejectCollabRequest.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+	// send notification
+
+	const collabAcceptedEvent = await ctx.db.createUserEvent({
+		type: 'COLLAB_REJECTED',
+		user: {connect: {id: request.requestee.id}},
+		metadata: {
+			userId: request.requestee.id,
+		},
+	});
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]:
```suggestion
		notifications: {
			create: {
				user: {connect: {id: request.requester.id}},
			},
		},
	});
```

File: resolvers/rejectCollabRequest.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]:
```suggestion
```

File: resolvers/rejectCollabRequest.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	await ctx.db.createNotification({
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]:
```suggestion
```

File: resolvers/rejectCollabRequest.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		userEvent: {connect: {id: collabAcceptedEvent.id}},
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]:
```suggestion
```

File: resolvers/rejectCollabRequest.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		user: {connect: {id: request.requester.id}},
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]:
```suggestion
```

File: resolvers/rejectCollabRequest.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	});
+
+	// Send email
+	try {
+		sendRejectCollabEmail(
+			{
+				email: request.requester.email,
+				meta: {userId: request.requester.id},
+				requesterName: formatFullName(
```
Comment (YoruNoHikage) [inline resolvers/rejectCollabRequest.js]: `formatName` also

File: resolvers/removeAssignmentToTask.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	});
+
+	const removeAssignmentToTaskEvent = await ctx.db.createUserEvent({
+		type: 'REMOVE_ASSIGNMENT_TO_TASK',
+		user: {connect: {id: getUserId(ctx)}},
+		metadata: {
+			itemId: taskId,
+		},
+	});
+
```
Comment (YoruNoHikage) [inline resolvers/removeAssignmentToTask.js]:
```suggestion
```

File: resolvers/removeAssignmentToTask.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+	await ctx.db.createNotification({
```
Comment (YoruNoHikage) [inline resolvers/removeAssignmentToTask.js]:
```suggestion
```

File: resolvers/requestCollab.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+	// Create collaboration
+	const newCollabRequest = await ctx.db.createCollabRequest({
+		requester: {connect: {id: getUserId(ctx)}},
+		requestee: {connect: {email: userEmail}},
+		status: 'PENDING',
+	});
+
+	// Create Requester user event
+	await ctx.db.createUserEvent({
+		type: 'COLLAB_REQUEST',
```
Comment (YoruNoHikage) [inline resolvers/requestCollab.js]:
```suggestion
		type: 'COLLAB_REQUESTED',
```

File: datamodel.prisma [hunk 10]
```diff
@@ -302,6 +317,7 @@ type Project {
 	deadline: DateTime
 	notifyActivityToCustomer: Boolean! @default(value: true)
 	attachments: [File!]! @relation(name: "ProjectAttachments")
+	collabLinkToProject: [User!]! @relation(name: "CollaboratorsProject")
```
Comment (YoruNoHikage) [inline datamodel.prisma]: Wouldn't collaborators be easier to understand?

File: resolvers/NotificationObject.js [hunk 11]
```diff
@@ -3,6 +3,12 @@ const NotificationObject = {
 		if (obj.token) {
 			return 'Project';
 		}
+		if (obj.email) {
+			return 'User';
+		}
+		if (obj.rejectedAt === null) {
```
[the remark this answers is not in the document]
Comment (FranzPoize) [reply resolvers/NotificationObject.js]: I wasn't sure what would be in requestee depending on the request.

Comment (FranzPoize) [inline resolvers/NotificationObject.js]:
```suggestion
		if (obj.requestee) {
```

File: resolvers/requestCollab.js [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+	// Send email
+	try {
+		sendRequestCollabEmail(
+			{
+				email: requestee.email,
+				meta: {userId: requestee.id},
+				requesteeName: formatName(requestee.firstName, requestee.lastName),
+				url: getAppUrl('/collaborators'),
+				user: formatName(undefined, user.firstName, user.lastName),
```
Comment (FranzPoize) [inline resolvers/requestCollab.js]:
```suggestion
				user: formatName(user.firstName, user.lastName),
```

**Pull request merged**